lib/src/Uncertainty/Model/ContinuousDistribution.cxx

```diff
@@ -1,145 +0,0 @@
- *
- *  This library is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU Lesser General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  This library is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU Lesser General Public License for more details.
- *
- *  You should have received a copy of the GNU Lesser General Public License
- *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
- *
- */
-#include <cmath>
-#include <cstdlib>
-
-#include "openturns/ContinuousDistribution.hxx"
-#include "openturns/Collection.hxx"
-#include "openturns/Distribution.hxx"
-#include "openturns/ResourceMap.hxx"
-#include "openturns/MethodBoundEvaluation.hxx"
-#include "openturns/GaussKronrod.hxx"
-
-BEGIN_NAMESPACE_OPENTURNS
-
-CLASSNAMEINIT(ContinuousDistribution)
-
-/* Default constructor */
-ContinuousDistribution::ContinuousDistribution()
-  : DistributionImplementation()
-{
-  // Nothing to do
-}
-
-/* Virtual constructor */
-ContinuousDistribution * ContinuousDistribution::clone() const
-{
-  return new ContinuousDistribution(*this);
-}
-
-/* Comparison operator */
-Bool ContinuousDistribution::operator ==(const ContinuousDistribution & other) const
-{
-  return (this == &other);
-}
-
-/* String converter */
-String ContinuousDistribution::__repr__() const
-{
-  OSS oss;
-  oss << "class=" << ContinuousDistribution::GetClassName();
-  return oss;
-}
-
-/* Get the PDF of the distribution */
-Scalar ContinuousDistribution::computePDF(const Point & ) const
-{
-  throw NotYetImplementedException(HERE) << "In ContinuousDistribution::computePDF(const Point & point) const";
-}
-
-/* Get the CDF of the distribution */
-Scalar ContinuousDistribution::computeCDF(const Point & point) const
-{
-  const Interval interval(getRange().getLowerBound(), point);
-  LOGINFO(OSS() << "In ContinuousDistribution::computeCDF, using computeProbabilityContinuous(), interval=" << interval.__str__());
-  return computeProbabilityContinuous(interval);
-}
-
-/* Get the survival function of the distribution */
-Scalar ContinuousDistribution::computeSurvivalFunction(const Point & point) const
-{
-  const UnsignedInteger dimension = getDimension();
-  if (dimension == 1) return computeComplementaryCDF(point);
-  const Point lowerBounds(getRange().getLowerBound());
-  const Point upperBounds(getRange().getUpperBound());
-  Bool allOutside = true;
-  for (UnsignedInteger i = 0; i < dimension; ++i)
-  {
-    if (point[i] >= upperBounds[i]) return 0.0;
-    allOutside &= (point[i] <= lowerBounds[i]);
-  }
-  if (allOutside) return 1.0;
-  const Scalar survival = computeProbability(Interval(point, getRange().getUpperBound()));
-  return survival;
```

Comment:
@sofianehaddad ContinuousDistribution::computeSurvivalFunction is not lost, because its code is in DistributionImplementation::computeSurvivalFunction : 

see https://github.com/openturns/openturns/blob/97dbfa5fa03878c9a350f21db8ad3c0895a47490/lib/src/Uncertainty/Model/DistributionImplementation.cxx#L776 and previous lines